### PR TITLE
AsmHighlighter Fix

### PR DESCRIPTION
**Pull request by Shelldll**
Hi! I've forked and modified your AsmHighlighter so that it now correctly highlights the syntax such as registers and directives. The issues was within the way that it was attempting to get the color index in the token switch statement. I've since updated not only that, but the references to some newer assemblies for Visual Studio 2015. As far as I can tell, it works pretty well now.

### CODE REVIEW

**Comment by Trass3r** [comment]
What exactly was the issue?

**Comment by Shelldll** [comment]
Only the instructions (mov, add, sub etc), strings, and numbers were being highlighted. I tested it with both VS2015 and VS2013 and neither had proper highlighting. This was because of the indices that were made in the AsmHighlighterColorableItem weren't matching up with the indices that were being assigned to the different tokens, resulting in no syntax highlighting. This fixes it. Also, I did find however that this does not work on VS2013 now as I had to update some references to use VS2015 so it would compile in VS2015, so there would be some compatibility issues unless recompiled on VS2013 with updated references I would assume. Also, I forgot that it does not commit the bin/debug folder so you may want to edit that change out of the Readme.md file I edited if you decide to pull.

**Comment by Trass3r** [comment]
Binaries shouldn't be put into git anyway.
For me it was quite strange. I only got partially working highlighting, but in the experimental instance it seemed to work better.

**Comment by Shelldll** [comment]
Yeah, I'm aware. Just a lapse of thought on my part, probably because I don't use github much. I believe I did notice the exp instance seemed to work better but I believe it should work on a default instance pretty well now. I'm able to change highlighting colors and all that now and seems to cover that vast majority of the items in the map file. Only issue now would be compatibility with VS2013 as far as I'm aware.

File: AsmHighlighter/AsmHighlighter.csproj [hunk 1]
```diff
@@ -58,7 +58,7 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="EnvDTE, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-      <EmbedInteropTypes>True</EmbedInteropTypes>
+      <EmbedInteropTypes>False</EmbedInteropTypes>
```
Comment (Trass3r) [inline AsmHighlighter/AsmHighlighter.csproj:61]: You missed the others.

File: AsmHighlighter/AsmHighlighterFormatHelper.cs [hunk 2]
```diff
@@ -340,9 +340,9 @@ public static List<EditSpan> ReformatCode(IVsTextLines pBuffer, TextSpan span, i
                         {
                             buffer = ManagedFasm.Assemble(codeToAssemble);
                         }
-                        catch (Exception ex)
+                        catch (Exception)
                         {
-                            // Unable to parse instruction... skip
+                            // ignored
```
Comment (Trass3r) [inline AsmHighlighter/AsmHighlighterFormatHelper.cs:345]: why replace the comment?

File: AsmHighlighter/AsmHighlighterLanguageService.cs [hunk 3]
```diff
@@ -44,24 +44,16 @@ public AsmHighlighterLanguageService()
...
-//				new AsmHighlighterColorableItem("ASM Comment", COLORINDEX.CI_DARKGREEN, COLORINDEX.CI_USERTEXT_BK),
-//				new AsmHighlighterColorableItem("ASM Identifier", COLORINDEX.CI_SYSPLAINTEXT_FG, COLORINDEX.CI_USERTEXT_BK),
-//				new AsmHighlighterColorableItem("ASM String", COLORINDEX.CI_RED, COLORINDEX.CI_USERTEXT_BK),
-//				new AsmHighlighterColorableItem("ASM Number", COLORINDEX.CI_DARKBLUE, COLORINDEX.CI_USERTEXT_BK),
-				new AsmHighlighterColorableItem("ASM Register", COLORINDEX.CI_MAROON, COLORINDEX.CI_USERTEXT_BK, FONTFLAGS.FF_BOLD),
-				new AsmHighlighterColorableItem("ASM FpuInstruction", COLORINDEX.CI_AQUAMARINE, COLORINDEX.CI_USERTEXT_BK),
-				new AsmHighlighterColorableItem("ASM Directive", COLORINDEX.CI_PURPLE, COLORINDEX.CI_USERTEXT_BK),
+				new AsmHighlighterColorableItem("Text", COLORINDEX.CI_USERTEXT_FG, COLORINDEX.CI_USERTEXT_BK),
+                new AsmHighlighterColorableItem("ASM Directive", COLORINDEX.CI_BLUE, COLORINDEX.CI_USERTEXT_BK),
+                new AsmHighlighterColorableItem("ASM Register", COLORINDEX.CI_AQUAMARINE, COLORINDEX.CI_USERTEXT_BK, FONTFLAGS.FF_BOLD),
```
Comment (Trass3r) [inline AsmHighlighter/AsmHighlighterLanguageService.cs:49]: whitespace inconsistencies

File: AsmHighlighter/AsmHighlighterScanner.cs [hunk 4]
```diff
@@ -25,34 +25,26 @@ namespace AsmHighlighter
...
     public class AsmHighlighterScanner : IScanner
     {
-        private Scanner lex;
-
         public AsmHighlighterScanner(IAsmHighlighterTokenProvider tokenProvider)
         {
-            lex = new Scanner();
-            lex.AsmHighlighterTokenProvider = tokenProvider;            
+            Lexer = new Scanner();
+            Lexer.AsmHighlighterTokenProvider = tokenProvider;
```
Comment (Trass3r) [inline AsmHighlighter/AsmHighlighterScanner.cs:31]: You mix cleanups and renames like this and above with the actual bugfix.

**Comment by Trass3r** [comment]
Compatibility with old versions of the IDE is not of concern. Extensions should still work when older toolsets are used though. But in this case that shouldn't matter.

**Comment by Trass3r** [comment]
Interestingly it seems like the highlighting suddenly works better with 2015 Update 2.

**Comment by Trass3r** [comment]
In the end it looks like this is the only way to get VS to use custom colors properly:
https://github.com/Trass3r/AsmHighlighter/commit/28c7d118ef39c432d65994ee6456cdb37c6d05f2

**Pull request closed without merging**